### PR TITLE
feature: add create set by set template

### PR DESCRIPTION
**Pull request by rykerli**
- 新功能
    - CMDB插件根据集群模板创建集群
- 优化项
    - 无
- bug fix
    - 无

### CODE REVIEW

File: pipeline_plugins/components/collections/sites/open/cc/create_set_by_template/legacy.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
+# -*- coding: utf-8 -*-
```
Comment (homholueng) [inline pipeline_plugins/components/collections/sites/open/cc/create_set_by_template/legacy.py:1]: 新插件统一用 v1_0 版本

File: pipeline_plugins/components/collections/sites/open/cc/create_set_by_template/legacy.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+from functools import partial
+
+from django.utils import translation
+from django.utils.translation import ugettext_lazy as _
+
+from pipeline.component_framework.component import Component
+from pipeline.core.flow.activity import Service
+from pipeline.core.flow.io import StringItemSchema, ArrayItemSchema, IntItemSchema
+from pipeline_plugins.base.utils.inject import supplier_account_for_business
+from pipeline_plugins.components.collections.sites.open.cc.base import cc_format_tree_mode_id, \
```
Comment (homholueng) [inline pipeline_plugins/components/collections/sites/open/cc/create_set_by_template/legacy.py:22]: from xxx import (
a,
b,
c
)

File: pipeline_plugins/components/collections/sites/open/docs/cc/create_set_by_template/legacy.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+* `填参方式` 选择父实例的方式
+    * `手动输入`: 文本输入
+            * `父实例` 文本输入的父实例路径集合
+            * 换行区分路径，`>`区分层级
+            * example: `a>b>c\n   a>b`
+            * 已容错：冗余回车/空格
+    * `拓扑选择`: 树形组件上勾选
+
+ 
+ * `集群名称` 集群名称
```
Comment (homholueng) [inline pipeline_plugins/components/collections/sites/open/docs/cc/create_set_by_template/legacy.md:24]: 多了个空格

File: pipeline_plugins/tests/components/collections/sites/open/cc_test/cc_create_set_by_template/test_legacy.py [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+
+GET_CLIENT_BY_USER = \
+    "pipeline_plugins.components.collections.sites.open.cc.create_set_by_template.legacy.get_client_by_user"
+CC_GET_CLIENT_BY_USER = \
+    'pipeline_plugins.components.collections.sites.open.cc.base.get_client_by_user'
+CC_FORMAT_PROP_DATA = \
+    "pipeline_plugins.components.collections.sites.open.cc.create_set_by_template.legacy.cc_format_prop_data"
+
+
+# 根据prop_i确定CC_FORMAT_PROP_DATA接口的返回值
```
Comment (homholueng) [inline pipeline_plugins/tests/components/collections/sites/open/cc_test/cc_create_set_by_template/test_legacy.py:61]: prop_id

File: pipeline_plugins/components/collections/sites/open/cc/create_set_by_template/legacy.py [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+        bk_set_env = cc_format_prop_data(
+            executor, "set", "bk_set_env", parent_data.get_one_of_inputs("language"), supplier_account
+        )
+
+        if not bk_set_env["result"]:
+            data.set_outputs("ex_data", bk_set_env["message"])
+            return False
+
+        bk_service_status = cc_format_prop_data(
```
Comment (homholueng) [inline pipeline_plugins/components/collections/sites/open/cc/create_set_by_template/legacy.py:116]: bk_service_status 在后面没有使用到？

File: pipeline_plugins/components/collections/sites/open/cc/create_set_by_template/legacy.py [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+            )
+            if not cc_list_select_node_inst_id_return["result"]:
+                data.set_outputs("ex_data", cc_list_select_node_inst_id_return["message"])
+                return False
+            cc_set_parent_select = cc_list_select_node_inst_id_return["data"]
+        else:
+            data.set_outputs("ex_data", _("请选择填参方式"))
+            return False
+
+        bk_set_env = cc_format_prop_data(
```
Comment (homholueng) [inline pipeline_plugins/components/collections/sites/open/cc/create_set_by_template/legacy.py:108]: bk_set_env 在后面没有使用到？

File: pipeline_plugins/tests/components/collections/sites/open/cc_test/cc_create_set_by_template/test_legacy.py [hunk 7]
```diff
@@ -0,0 +1,279 @@
...
+
+
+class CCCreateSetByTemplateComponentTest(TestCase, ComponentTestMixin):
+    def component_cls(self):
+        return CCCreateSetBySetTemplateComponent
+
+    def cases(self):
+        return [
+            SELECT_BY_TEXT_SUCCESS_CASE,
+            SELECT_BY_TEXT_ERROR_LEVEL_FAIL_CAST,
```
Comment (homholueng) [inline pipeline_plugins/tests/components/collections/sites/open/cc_test/cc_create_set_by_template/test_legacy.py:35]: CASE

File: pipeline_plugins/tests/components/collections/sites/open/cc_test/cc_create_set_by_template/test_legacy.py [hunk 8]
```diff
@@ -0,0 +1,279 @@
...
+            SELECT_BY_TEXT_ERROR_PATH_FAIL_CAST,
```
Comment (homholueng) [inline pipeline_plugins/tests/components/collections/sites/open/cc_test/cc_create_set_by_template/test_legacy.py:36]: CASE

File: pipeline_plugins/components/collections/sites/open/cc/create_set_by_template/legacy.py (hunk 6 above)
Comment (rykerli) [reply pipeline_plugins/components/collections/sites/open/cc/create_set_by_template/legacy.py:108]: 紧接着下面就用到了，会判断search_object_attribute是否调用成功

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/bk-sops?pullRequest=2265) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/bk-sops?pullRequest=2265) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/bk-sops?pullRequest=2265) it.</sub>

**Pull request merged**